modules/build/src/main/scala/scala/build/options/validation/BuildOptionsRule.scala

```diff
@@ -0,0 +1,34 @@
+package scala.build.options.validation
+
+import scala.build.Position
+import scala.build.errors.BuildException
+import scala.build.options.BuildOptions
+
+trait BuildOptionsRule {
+  def validate(options: BuildOptions): Either[BuildException, Unit]
+}
+
+object BuildOptionsRule {
+  def validateAll(options: BuildOptions): Either[BuildException, Unit] =
+    List(JvmOptionsForNonJvmBuild).map(_.validate(options)).partition(_.isLeft) match {
+      case (Nil, _)  => Right(())
+      case (some, _) => some.head
```

Comment:
This discard all validation exceptions but the first one. Better keep them all via
```suggestion
      case (some, _) => Left(CompositeBuildException(some.flatMap(_.left.toSeq)))
```